driver/src/contracts/stablex_contract.rs

```diff
@@ -59,12 +59,86 @@ impl StableXContract for StableXContractImpl {
 
     fn submit_solution(
         &self,
-        _batch_index: U256,
-        _orders: Vec<Order>,
-        _solution: Solution,
+        batch_index: U256,
+        orders: Vec<Order>,
+        solution: Solution,
     ) -> Result<()> {
-        unimplemented!();
+        let account = self
+            .base
+            .account_with_sufficient_balance()
+            .ok_or("Not enough balance to send Txs")?;
+
+        let (owners, order_ids, volumes, prices, token_ids_for_price) =
```

Comment:
nit: would it make sense to have one method to get owners, orders, volumes and a separate one for prices and their indices (given they are disjunct in logic)?

Reply:
Yea, sure! I like it!

Reply:
Although, separating this into two functions will result in an overhaul of the unit tests (i.e. testing them each). Do is this a change that could be made in the future or would you suggest it be done here?

Reply:
As you feel. It's really optional anyways.

Reply:
Ok, so I made these changes and they drastically reduced the complexity of the unit testing. Unfortunately, I would suspect that (eventually) we wouldn't require the price information to be "converted" since this is ultimately not scalable in the number of tokens. Anyway. Let me know what you think and I believe this is ready for merge!